src/transports/httpclient.c

```diff
@@ -305,26 +408,201 @@ const char *name_for_method(git_http_method method)
 	return NULL;
 }
 
+/*
+ * Find the scheme that is suitable for the given credentials, based on the
+ * server's auth challenges.
+ */
+static bool best_scheme_and_challenge(
+	git_http_auth_scheme **scheme_out,
+	const char **challenge_out,
+	git_vector *challenges,
+	git_cred *credentials)
+{
+	const char *challenge;
+	size_t i, j;
+
+	for (i = 0; i < ARRAY_SIZE(auth_schemes); i++) {
+		git_vector_foreach(challenges, j, challenge) {
+			git_http_auth_scheme *scheme = &auth_schemes[i];
+
+			if (challenge_matches_scheme(challenge, scheme) &&
+			    !!(scheme->credtypes & credentials->credtype)) {
```

Comment:
No need for the double negation, you only want to know if the bit is set but do not care for its value